core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -72,6 +72,7 @@
     public static final String QUERY_MAX_EXECUTION_TIME = "query_max_execution_time";
     public static final String QUERY_MAX_PLANNING_TIME = "query_max_planning_time";
     public static final String QUERY_MAX_RUN_TIME = "query_max_run_time";
+    public static final String QUERY_REPORTED_RULE_STATS_LIMIT = "query_reported_rule_stats_limit";
```

Comment:
Since query info is not a user visible change, therefore I don't think we should expose session property for `query_reported_rule_stats_limit`. Just feature config is enough